--- GodotAppOriginal.java	2025-08-07 12:07:46.000000000 +0530
+++ GodotApp.java	2025-08-07 19:36:45.992086899 +0530
@@ -34,6 +34,8 @@
 import org.godotengine.godot.GodotActivity;
 
 import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
 import android.util.Log;
 
 import androidx.activity.EdgeToEdge;
@@ -44,6 +46,7 @@
  * Feel free to extend and modify this class for your custom logic.
  */
 public class GodotApp extends GodotActivity {
+	private boolean isAppReady = false;
 	static {
 		// .NET libraries.
 		if (BuildConfig.FLAVOR.equals("mono")) {
@@ -67,9 +70,10 @@
 
 	@Override
 	public void onCreate(Bundle savedInstanceState) {
-		SplashScreen.installSplashScreen(this);
+		SplashScreen splashScreen = SplashScreen.installSplashScreen(this);
 		EdgeToEdge.enable(this);
 		super.onCreate(savedInstanceState);
+		splashScreen.setKeepOnScreenCondition(() -> !isAppReady);
 	}
 
 	@Override
@@ -82,5 +86,8 @@
 	public void onGodotMainLoopStarted() {
 		super.onGodotMainLoopStarted();
 		runOnUiThread(updateWindowAppearance);
+		new Handler(Looper.getMainLooper()).postDelayed(() -> {
+			isAppReady = true;
+		}, 300);
 	}
 }
